.circleci/workflows.yml

```diff
@@ -0,0 +1,1094 @@
+  deploy-to-pypi:
+    docker: *docker
+    steps:
+      - checkout
+      - run:
+          name: Install deployment tools
+          command: |
+            pip install --upgrade setuptools wheel twine
+      - run:
+          name: Create the distribution files
+          command: |
+            python -m build --sdist --wheel
+      - run:
+          name: Upload to PyPI
+          command: |
+            # Relies on the TWINE_USERNAME and TWINE_PASSWORD environment variables
+            # For more on twine, see:
+            #   https://twine.readthedocs.io/en/latest/
+            twine upload --skip-existing dist/*
+
+workflows:
+  version: 2
+  build:
+    jobs: &build_jobs
+      - manual-trigger-required-for-fork
+      - build
+      - verify-requirements
+      - verify-format-yaml
+      - deploy-to-pypi:
+          name: Deploy bqetl to PyPI
+          requires:
+            - build
+          filters:
+            tags:
+              only: /[0-9]{4}.[0-9]{1,2}.[0-9]+/  # Calver: YYYY.M.MINOR
+            branches:
+              # Ignore all branches; this workflow should only run for tags.
+              ignore: /.*/
+
+  ### Validate SQL
+  validate-sql:
+    when:
+      or:
+        - << pipeline.parameters.validate-sql >>
+        - << pipeline.parameters.validate-routines >>
+    jobs:
+      - verify-format-sql
+      - deploy-changes-to-stage:
+          requires:
+            - generate-sql
+      - dry-run-sql:
+          requires:
+            - deploy-changes-to-stage
+      - test-sql:
+          requires:
+            - deploy-changes-to-stage
+      - validate-metadata:
+          requires:
+            - deploy-changes-to-stage
+      - validate-dags:
+          requires:
+            - generate-dags
+      - validate-views:
+          requires:
+            - deploy-changes-to-stage
+      - generate-sql
+      - main-generate-sql-and-dags:
+          filters:
+            branches:
+              ignore: main
+      - generate-diff:
+          requires:
+            - generate-dags
+            - main-generate-sql-and-dags
+          filters:
+            branches:
+              ignore: main
+      - post-diff:
+          requires:
+            - generate-diff
+          filters:
+            branches:
+              ignore: main
+      - generate-dags:
+          requires:
+            - generate-sql
+      - docs:
+          requires:
+            - generate-sql
+          filters:
+            branches:
+              only: main
+      - push-generated-sql:
+          requires:
+            - validate-dags
+          filters:
+            branches:
+              only:
+                - main
+      - sync-dags-repo:
+          name: 🔃 Synchronize bigquery-etl submodule
+          repo-to-sync: ${CIRCLE_PROJECT_REPONAME}
+          target-branch: generated-sql
+          requires:
+            - push-generated-sql
+          filters:
+            branches:
+              only:
+                - main
```

Comment:
I think this and the `ignore: main` above can be removed since this workflow doesn't run on main, right?